src/infra/core/src/compacting_service_impl.rs

```diff
@@ -515,4 +516,42 @@ impl CompactingService for CompactingServiceImpl {
             }
         }
     }
+
+    async fn compact_multi(
+        &self,
+        dataset_refs: Vec<DatasetRef>,
+        options: CompactingOptions,
+        multi_listener: Option<Arc<dyn CompactingMultiListener>>,
+    ) -> Vec<CompactingResponse> {
+        let dataset_handles: Vec<_> = match filter_datasets_by_local_pattern(
+            self.dataset_repo.as_ref(),
+            dataset_refs
+                .into_iter()
+                .map(DatasetRefPattern::Ref)
+                .collect(),
+        )
+        .try_collect()
+        .await
```

Comment:
optional: extract `filter_datasets_by_local_pattern(...).try_collect().await` to a local let